DevelopGitHub/GitHubExtension/GitHubExtension.LocalizationTool/Translate/Translator.cs

```diff
@@ -0,0 +1,186 @@
+            "lang=";
+
+        private readonly MainWindow mainWindow;
+
+        public Translator(MainWindow mainWindow)
+        {
+            this.mainWindow = mainWindow;
+        }
+
+        private ObservableCollection<TranslationDataRow> TranslationData
+        {
+            get
+            {
+                return mainWindow.TranslationData;
+            }
+        }
+
+        public static Lang GetLang(string lang)
+        {
+            switch (lang)
+            {
+                case "uk":
+                case "UK":
+                case "uk-UK":
+                case "uk-UK.json":
+                    return Lang.Uk;
+                case "us":
+                case "US":
+                case "en-Us":
+                case "en-Us.json":
+                    return Lang.Us;
+                case "ru":
+                case "RU":
+                case "ru-RU":
+                case "ru-RU.json":
+                    return Lang.Ru;
+                default:
+                    throw new Exception("Incorrect Lang in Method GetLang");
+            }
+        }
+
+        public static string GetLang(Lang lang)
+        {
+            switch (lang)
+            {
+                case Lang.Uk:
+                    return "uk-UK";
+                case Lang.Us:
+                    return "en-US";
+                case Lang.Ru:
+                    return "ru-RU";
+                default:
+                    throw new Exception("Incorrect Lang in Method GetLang");
+            }
+        }
+
+        public static string GetFileName(Lang lang)
+        {
+            switch (lang)
+            {
+                case Lang.Uk:
+                    return "uk-UK.json";
+                case Lang.Us:
+                    return "en-US.json";
+                case Lang.Ru:
+                    return "ru-RU.json";
+                default:
+                    throw new Exception("Incorrect Lang in Method GetLang");
+            }
+        }
+
+        public void AddNewRowToTranslationData(Lang language, KeyValuePair<string, JToken> element)
+        {
+            var line = new TranslationDataRow(element.Key);
+            line[language] = element.Value.ToString();
+            TranslationData.Add(line);
+        }
+
+        public async void WebTranslate(string sourceLanguage, string targetLanguage)
+        {
+            if (sourceLanguage.Equals(targetLanguage))
+            {
+                return;
+            }
+
+            Lang sourceLanguageEnum;
+            Lang targetLanguageEnum;
+            TargetLanguageEnum(ref sourceLanguage, ref targetLanguage, out sourceLanguageEnum, out targetLanguageEnum);
+
+            StringBuilder textToTranslate = GenerateTextParameter(sourceLanguageEnum);
+
+            var result = Task<JObject>.Factory.StartNew(() => GetTranslationFromYandexApi(sourceLanguage, targetLanguage, textToTranslate));
```

Comment:
I used async because response waiting can take a lot of time
User can continue filling the data into table while this method is waiting response from Yandex translate server